File: src/demos/ps/dungeons/Dezo_cave2.java
package demos.ps.dungeons;

import demos.ps.PSDungeon;
import demos.ps.oo.PSGame;
import demos.ps.oo.PSGame.Planet;
import demos.ps.oo.PSLibEnemy.PS1Enemy;

public class Dezo_cave2 {
	
	public static void startmap() {
		PSDungeon dungeon = PSGame.currentDungeon;
		dungeon.setRandomEnemies(0, new PS1Enemy[]{PS1Enemy.ZOMBIE});
		dungeon.setFixedEnemies(0, new PS1Enemy[]{PS1Enemy.ZOMBIE, PS1Enemy.BATALION, PS1Enemy.ZOMBIE, PS1Enemy.ZOMBIE});
		dungeon.startDungeon();
	}
	
	public static void exit() {
		PSGame.mapswitch(Planet.DEZORIS,159,86);
	}
	public static void nextArea() {
		PSGame.mapswitch(Planet.DEZORIS,150,80);
	}
	
}

File: src/demos/ps/dungeons/Dezo_cave2.ts
/**
 * Dezo_cave2 - Dungeon Script
 * TypeScript port of Dezo_cave2.java
 */

import { PSGame } from '../PSGame';
import { Planet } from '../game/City';
import { PS1Enemy } from '../game/PSLibEnemy';
import { MainEngine } from '../../../core/MainEngine';

export class Dezo_cave2 {

  public static async startmap(): Promise<void> {
    MainEngine.setScriptContext(Dezo_cave2);
    const dungeon = PSGame.getCurrentDungeonInstance();
    if (dungeon) {
      dungeon.setRandomEnemies(0, [PS1Enemy.ZOMBIE]);
      dungeon.setFixedEnemies(0, [PS1Enemy.ZOMBIE, PS1Enemy.BATALION, PS1Enemy.ZOMBIE, PS1Enemy.ZOMBIE]);
      await dungeon.startDungeon();
    }
  }

  public static async exit(): Promise<void> {
    await PSGame.mapswitchToPlanet(Planet.DEZORIS, 159, 86);
  }

  public static async nextArea(): Promise<void> {
    await PSGame.mapswitchToPlanet(Planet.DEZORIS, 150, 80);
  }
}
